tst/EnergyPlus/unit/ZoneHVACEvaporativeCooler.unit.cc

```diff
@@ -0,0 +1,527 @@
+        "   ,                             !- Availability Manager List Name",
+        "   ZoneEvapCool OA Inlet,        !- Outdoor Air Inlet Node Name",
+        "   ZoneEvapCool Inlet Node,      !- Cooler Outlet Node Name",
+        "   ZoneEvapCool Relief Node,     !- Zone Relief Air Node Name",
+        "   Fan:OnOff,                    !- Supply Air Fan Object Type",
+        "   ZoneEvapCool Supply Fan,      !- Supply Air Fan Name",
+        "   1.0,                          !- Design Supply Air Flow Rate {m3/s}",
+        "   BlowThrough,                  !- Fan Placement",
+        "   ZoneTemperatureDeadbandOnOffCycling,  !- Cooler Unit Control Method",
+        "   1.0,                          !- Throttling Range Temperature Difference {deltaC}",
+        "   100.0,                        !- Cooling Load Control Threshold Heat Transfer Rate {W}",
+        "   EvaporativeCooler:Direct:CelDekPad,  !- First Evaporative Cooler Object Type",
+        "   Direct CelDekPad EvapCooler;  !- First Evaporative Cooler Object Name",
+
+        " Fan:OnOff,",
+        "    ZoneEvapCool Supply Fan,     !- Name",
+        "    ,                            !- Availability Schedule Name",
+        "    0.7,                         !- Fan Total Efficiency",
+        "    300.0,                       !- Pressure Rise {Pa}",
+        "    1.0,                         !- Maximum Flow Rate {m3/s}",
+        "    0.9,                         !- Motor Efficiency",
+        "    1.0,                         !- Motor In Airstream Fraction",
+        "    ZoneEvapCool OA Inlet,       !- Air Inlet Node Name",
+        "    ZoneEvapCool Fan outlet;     !- Air Outlet Node Name",
+
+        " EvaporativeCooler:Direct:CelDekPad,",
+        "    Direct CelDekPad EvapCooler, !- Name",
+        "    ,                            !- Availability Schedule Name",
+        "    0.6,                         !- Direct Pad Area {m2}",
+        "    0.17,                        !- Direct Pad Depth {m}",
+        "    55,                          !- Recirculating Water Pump Power Consumption {W}",
+        "    ZoneEvapCool Fan outlet,     !- Air Inlet Node Name",
+        "    ZoneEvapCool Inlet Node,     !- Air Outlet Node Name",
+        "    ;                            !- Control Type",
+
+    });
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    ScheduleManager::ProcessScheduleInput();
+    ScheduleManager::ScheduleInputProcessed = true;
+
+    Fans::GetFanInput();
+    ASSERT_FALSE(ErrorsFound);
+    EvaporativeCoolers::GetEvapInput();
+    ASSERT_FALSE(ErrorsFound);
+    EvaporativeCoolers::GetInputZoneEvaporativeCoolerUnit();
+    ASSERT_FALSE(ErrorsFound);
+
+    DataGlobals::BeginEnvrnFlag = true;
+    DataZoneEquipment::ZoneEquipInputsFilled = true;
+
+    auto &thisZoneEvapCooler(ZoneEvapUnit(UnitNum));
+
+    DataZoneEquipment::ZoneEquipConfig(1).ExhaustNode(1) = thisZoneEvapCooler.UnitReliefNodeNum;
+
+    DataLoopNode::Node.redimension(NumOfNodes);
```

Comment:
Hmm, redimension after `GetInputZoneEvaporativeCoolerUnit`? Not following why that should happen here.

Reply:
Required to include the zone related nodes defined in the EnergyPlusFixture::SetUp() [here](https://github.com/NREL/EnergyPlus/blob/166330897_Issue6707/tst/EnergyPlus/unit/ZoneHVACEvaporativeCooler.unit.cc#L99-L114);

Reply:
The name of the evap cooler is different from the base fixture and the zone inlet node name looks unique, so number of nodes will probably be different from what was expected based on the base fixture setup. But I agree that the need for reallocation of nodes is suspect. Didn't `GetInputZoneEvaporativeCoolerUnit` allocate new nodes as they were read in (e.g., DataZoneEquipment::ZoneEquipConfig(1).ZoneNode should be a valid node)? What was NumOfNodes compared to the allocated Node array size?

Reply:
Get input does not allocate all nodes required for zone connection. That is why I re-dimension the "Node" array.

Reply:
The number of nodes allocated during get input depends on the unit test but it was some where between 4 - 7, I think.  I reserved node number 9 and 10 for return and zone air node.